compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -896,10 +898,23 @@ trait Implicits { self: Typer =>
         case cand :: pending1 =>
           tryImplicit(cand) match {
             case fail: SearchFailure =>
-              if (fail.isAmbiguous) healAmbiguous(pending1, fail)
-              else rank(pending1, found, fail :: rfailures)
+              if (isNot)
```

Comment:
Wouldn't it be more efficient to check if we're looking for a `Not[T]` before doing the search, this way we know we only need to do a search for `T` once, whereas with the current implementation I guess the search will be done both for `Not.amb1` and `Not.amb2` ?

Reply:
In fact we do the search for `T` only once. if we fail at that, we immediately succeed for `Not[T]`.